File: Raising_Bacteria.py
# We are tasked with putting some bacteria in a box each day, and each night, the bacteria splits into 2 parts.
# We need to figure out how many bacteria should be put across days to see 'x' bacteria on a particular day.

def min_bacteria(n):
    # Count the number of set bits (1s) in the binary representation of n
    return bin(n).count('1')

n = int(input())

# Output the result
print(min_bacteria(n))

File: Raising_Bacteria.cpp
// each day we put some bacteria in a box and at night it splits into 2 parts
// we want to see x bacteria on a particular day
// initially box was empty
// find min number of bacteria to be put across the days
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;

int main(){
    ll x;
    cin >> x;

    ll lower_bound = 1;
    while(lower_bound <= x){
        lower_bound *= 2;
    }
    cout << 1 + x - (lower_bound / 2) << endl;
}